### PR TITLE
bump alibi-detect to 0.8.1 in adserver

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

We need to have up to date alibi-detect

**Which issue(s) this PR fixes**:

Fixes https://github.com/SeldonIO/seldon-core/issues/3864

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 131dd4d046b8ebe2efe8729aeddaf295adf54904 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3871-notebooks-mtnjf) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 8d18717b99b2960dbc23a0b6f422362d568590fb | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3871-integration-hrhfk) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
integration / notebook tests passed 
image
but now I see failure in unit tests that was not present before

**Comment by RafalSkolasinski** [comment]
It had to be some flakiness in a single run during installation of dependencies, all good again 
image

File: components/alibi-detect-server/pyproject.toml
Comment (sakoush) [inline components/alibi-detect-server/pyproject.toml:14]: This is slightly out of date, and `alibi-detect` is compatible with tf version up to 2.7, worth upgrading?

Can be left as a separate PR though.

File: components/alibi-detect-server/Dockerfile
Comment (sakoush) [inline components/alibi-detect-server/Dockerfile:74]: is this something introduced with new alibi-detect version?

**Review by sakoush — APPROVED** [review]
LGTM.

Do we need to regenerate artifacts?

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *[sakoush](https://github.com/SeldonIO/seldon-core/pull/3871#pullrequestreview-865950789 "Approved")*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/alibi-detect-server/Dockerfile
Comment (RafalSkolasinski) [reply components/alibi-detect-server/Dockerfile:74]: yes, there is new dependency on numba and numba's default cache directory is not writable which lead to failures

File: components/alibi-detect-server/pyproject.toml
Comment (RafalSkolasinski) [reply components/alibi-detect-server/pyproject.toml:14]: Yeah, this makes sense. I can try and see where it lead me - if no issues could go in here

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3871#issuecomment-1028109420 "Approved")*, *[sakoush](https://github.com/SeldonIO/seldon-core/pull/3871#pullrequestreview-865950789 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
